### ISSUE TITLE
Mark inactive elements for review when using bad ARIA

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3639]
While rare, on occasion improper use of ARIA ends up not causing any significant accessibility barriers / WCAG violations. As far as there is a pattern to it, the false positives I've seen come from axe-core's ARIA rules tend to be are most common on static elements; people making up non-existing semantics.

So to cut the risk of false positives on ARIA rules I propose that for the ARIA rules we report `incomplete` when all the following conditions are met:

- The element has no widget-role; and
- The element is not in the tab order; and
- The element has no global ARIA attributes (like aria-label, aria-live, etc.)

This is similar to the logic we use to determine presentational role conflict. I propose we add this exception to the following axe-core rules:

- aria-allowed-attr
- aria-roles
- aria-roledescription
- aria-valid-attr
- aria-valid-attr-value

### Examples

```html
<!-- fail -->
<button role="magic-button"></button>
<div aria-fake-attribute="bad-value" tabindex="0"></div>
<div aria-live="hidden" aria-label="button!"></div>

<!-- incomplete -->
<div role="made-up-navbar"></div>
<div role="main" aria-description="Main navigation"></div>
<div aria-roledescription="super-search" tabindex="-1"></div>
```

### Alternative 1: Best Practice

One way to avoid false positives for WCAG would be not to report ARIA violations as violations of WCAG at all. This is what tools like WAVE do for example. The problem with it is that 95% of the time bad ARIA really does cause a violation of WCAG, so not mapping it to WCAG seems problematic too.

### Alternative 2: Always Incomplete

Axe-core does have some rules that only report incomplete. If we were to do the same for these four rules, we'd avoid this problem. The downside is that it creates a lot more "needs review" items coming from axe-core. People tend not to review these, and so high probability accessibility issues will get ignored. We would also not be able to get these rules listed as "consistent" with ACT.
